src/index.ts

```diff
@@ -248,15 +252,60 @@ export class MayanRoute<N extends Network>
       const rpc = await this.request.fromChain.getRpc();
       const txs: TransactionId[] = [];
       if (this.request.fromChain.chain === "Solana") {
-        const swapResult = await swapFromSolana(
-          quote.details!,
-          originAddress,
-          destinationAddress,
-          this.referrerAddress(),
-          mayanSolanaSigner(signer),
-          rpc
-        );
-        txs.push({ chain: "Solana", txid: swapResult.signature });
+        const { instructions, signers, lookupTables } =
+          await createSwapFromSolanaInstructions(
+            quote.details!,
+            originAddress,
+            destinationAddress,
+            this.referrerAddress(),
+            rpc
+          );
+        const swapper = new PublicKey(originAddress);
+
+        const feePayer = quote.details!.gasless
```

Comment:
The `fetchQuote` function used in line 175 correctly sets `gasless=false` by default. This ensures that the swap mode is assumed to be non-gasless, and the `swapper` is always used as the `feePayer`. Therefore, you don't need to consider the possibility of a gasless quote.